frontend/src/store/modules/BooksModule.ts

```diff
@@ -36,11 +37,25 @@ class BooksModule extends VuexModule {
   }
 
   @Action({rawError: true})
-  public create(request: BookForm): Promise<Book> {
+  public fetchSingle(barcode: string): Promise<Book> {
+    return new Promise((resolve, reject) => {
+      BooksAPI.single(barcode)
+        .then((response: any) => {
+          resolve();
+        })
+        .catch((error: any) => {
+          this.setfailure(error);
+          reject(error);
+        });
+    });
+  }
+
+  @Action({rawError: true})
+  public create(request: any) {
     return new Promise((resolve, reject) => {
       BooksAPI.create(request)
-        .then((response: Book) => {
-          this.setBook(response);
+        .then((response: any) => {
```

Comment:
Shouldn't this be of type Book?
```suggestion
        .then((response: Book) => {
```